packages/app/src/providers/utils.ts

```diff
@@ -81,6 +81,17 @@ export function convertDtype(dtype: string): DType {
   }
 }
 
+export function flattenValue(
+  value: unknown,
+  entity: Dataset<ArrayShape>,
+  selection?: string
+) {
+  assertArray(value);
```

Comment:
This will throw if we ask the API to select a scalar, so at least we'll know (in the unlikely event we write code that does this inadvertently).

Reply:
I understand that given our usecases, we will never get a scalar from an array dataset.

Still it bothers me to prevent it for the following reasons:

- Here and in `assertDatasetValue` usage locations, we have access to `selection`. So we could assert according to the `selection`. 
- The type inference in `useDatasetValue` could make use of `dimMapping` to have a `Value<D>` corresponding to the gotten slice.

Anyway, this will do for now.

Reply:
Sure, it's all doable, but adding this level of type safety to `useDatasetValue` means that we then have to assert that the returned value is not scalar inside the relevant containers, which leads to exactly the same result as the assertion that the value returned by the provider is an array regardless of `selection`.

If anything, it would make more sense to prevent `useDatasetValue` from being called with a `dimMapping` that only contains numbers.

Reply:
Either way, we still have implicit coupling between `useDatasetValue` and the provider APIs' `getValue` methods... Perhaps a better solution would be to rethink the signature of `getValue` to make this coupling more explicit - e.g. by letting it accept a dataset rather than an entity and an optional dimension mapping array with at least one axis in it.